### ISSUE TITLE
[Fatal] PHP Fatal error: Uncaught RuntimeException: Unable to claim actions | ActionScheduler_DBStore.php:944

### ISSUE THREAD

**Issue by nigeljamesstevenson** [woothemes/woocommerce#52157]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

We observed the following fatal error in relation to WooCommerce 9.3.3

[Link to code](https://github.com/woocommerce/action-scheduler/blob/682caac422df2976dc17edb1abfe6cd2d08e2cdf/classes/data-stores/ActionScheduler_DBStore.php#L944)

Stack trace:

```

PHP Fatal error:  Uncaught RuntimeException: Unable to claim actions. Database error: Lock wait timeout exceeded; try restarting transaction. in /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php:944
Stack trace:
#0 /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php(801): ActionScheduler_DBStore->claim_actions(2135062, 25, NULL, '', '')
woocommerce/woocommerce#1 /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/data-stores/ActionScheduler_HybridStore.php(221): ActionScheduler_DBStore->stake_claim(25, NULL, Array, '')
woocommerce/woocommerce#2 /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(160): ActionScheduler_HybridStore->stake_claim(25)
woocommerce/woocommerce#3 /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(139): ActionScheduler_QueueRunner->do_batch(25, 'Async Request')
woocommerce/woocommerce#4 /../core/6.6.2/wp-includes/class-wp-hook.php(324): ActionScheduler_QueueRunner->run('Async Request')
woocommerce/woocommerce#5 /../core/6.6.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
woocommerce/woocommerce#6 /../core/6.6.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
woocommerce/woocommerce#7 /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/ActionScheduler_AsyncRequest_QueueRunner.php(52): do_action('action_schedule...', 'Async Request')
woocommerce/woocommerce#8 /srv/htdocs/wp-content/plugins/astra-sites/inc/lib/starter-templates-importer/importer/helpers/wp-async-request.php(154): ActionScheduler_AsyncRequest_QueueRunner->handle()
woocommerce/woocommerce#9 /../core/6.6.2/wp-includes/class-wp-hook.php(324): WP_Async_Request->maybe_handle('')
woocommerce/woocommerce#10 /../core/6.6.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
woocommerce/woocommerce#11 /../core/6.6.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
woocommerce/woocommerce#12 /../core/6.6.2/wp-admin/admin-ajax.php(207): do_action('wp_ajax_nopriv_...')
woocommerce/woocommerce#13 {main}
  thrown in /../plugins/woocommerce/9.3.3/packages/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php on line 944
``` 

### Expected behavior

No fatal errors when attempting to claim actions

### Actual behavior

Fatal errors when attempting to claim actions

### Steps to reproduce

We don't have steps to reproduce, but observed 750 instances of this problem on Fri Oct 11, 1200 UTC to 2359 UTC on ~ 10 sites. Hopefully, someone can dig in and apply a defensive check if appropriate. - To be updated after investigation.

### WordPress Environment

N/A

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by albarin** [comment]
May be related to https://github.com/woocommerce/action-scheduler/issues/1262

**Comment by jorgeatorres** [comment]
See https://github.com/woocommerce/action-scheduler/issues/1249.
